src/Twilio/AuthStrategies/TokenAuthStrategy.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Threading;
+using Twilio.Http.BearerToken;
+using Twilio.Exceptions;
+
+#if !NET35
+using System.IdentityModel.Tokens.Jwt;
+using System.Threading.Tasks;
+#endif
+
+#if NET35
+using Twilio.Http.Net35;
+using System.Collections.Generic;
+using System.Text;
+using System.Web.Script.Serialization;
+#endif
+
+namespace Twilio.AuthStrategies
+{
+    public class TokenAuthStrategy : AuthStrategy
+    {
+        private string token;
+        private TokenManager tokenManager;
+
+
+        public TokenAuthStrategy(TokenManager tokenManager)
+        {
+            this.tokenManager = tokenManager ?? throw new ArgumentNullException(nameof(tokenManager));
+        }
+
+        public override string GetAuthString()
+        {
+            FetchToken();
+            return $"Bearer {token}";
+        }
+
+        public override bool RequiresAuthentication()
+        {
+            return true;
+        }
+
+        // Token-specific refresh logic
+        private void FetchToken()
+        {
+            if (string.IsNullOrEmpty(token) || tokenExpired(token))
+            {
+                lock (typeof(TokenAuthStrategy))
+                {
+                    if (string.IsNullOrEmpty(token) || tokenExpired(token))
+                    {
+                        token = tokenManager.fetchAccessToken();
+                    }
+                }
+            }
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj == null || GetType() != obj.GetType()) return false;
+            var that = (TokenAuthStrategy)obj;
+            return token == that.token && tokenManager.Equals(that.tokenManager);
+        }
+
+
+        public bool tokenExpired(String accessToken){
+            #if NET35
+            return IsTokenExpired(accessToken);
+            #else
+            return isTokenExpired(accessToken);
+            #endif
```

Comment:
Do we have any function that works in both old and new framework? instead of making two different functions, if we can use single one to do it?

Reply:
we cannot use single function to do it, whatever works in 3.5 is deprecated in later versions and vice versa